src/screens/Ticketing/Purchase/utils.ts

```diff
@@ -14,21 +23,42 @@ export type PurchaseFlow = {
    * ticket.
    */
   travelDateSelectionEnabled: boolean;
+
+  /**
+   * Whether the customer should be able to select between a set of durations
+   * for the ticket, based on `fareproduct.durationDays`.
+   */
+  durationSelectionEnabled: boolean;
 };
 
 export const getPurchaseFlow = (
   product: PreassignedFareProduct,
 ): PurchaseFlow => {
-  if (product.type === 'period' || product.type === 'hour24') {
-    return {
-      travellerSelectionMode: 'single',
-      travelDateSelectionEnabled: true,
-    };
-  } else {
-    return {
-      travellerSelectionMode: 'multiple',
-      travelDateSelectionEnabled: false,
-    };
+  switch (product.type) {
+    case 'period':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: true,
+        durationSelectionEnabled: true,
+      };
+    case 'hour24':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: true,
+        durationSelectionEnabled: false,
+      };
+    case 'single':
+      return {
+        travellerSelectionMode: 'multiple',
+        travelDateSelectionEnabled: false,
+        durationSelectionEnabled: false,
+      };
+    case 'carnet':
+      return {
+        travellerSelectionMode: 'single',
+        travelDateSelectionEnabled: false,
+        durationSelectionEnabled: false,
+      };
```

Comment:
Carnet is not available for purchase in the app, but added for completeness sake, instead of having a fallback. Also means the same function could be used in the webshop.